Fix issue where enclosing class references are not correctly updated

jgallimore opened the pull request:
There are some cases where a class A encloses another class B, and B makes a reference to A, the reference may not be correctly re-written.

jgallimore commented:
I'll dig into the JDK14/15 test failures here.

jgallimore commented:
This relates to issue #45 .

bjhargrave commented:
The fix looks fine, albeit with a lot of whitespace changes. (I need to do a source code format over the repo to normalize the whitespace.)

Once you sort out the Java 14+ test failures, please squash all the commits into a single commit. Thanks.

bjhargrave commented:
> I need to do a source code format over the repo to normalize the whitespace.

I have opened #60 to format the source code into a "normalized" state.

jgallimore commented:
Thanks for the review! The test failure was down to the constants being different between JDK versions. Newer versions of the JDK create an additional NameAndType constant for Ltransformer/test/data/xformme/Sample_OuterClass;this$0, and earlier JDKs do not. I have specifically commented this in the test to avoid confusion, and have extended the assertions to check the constant pool to make sure there are no references to the old package.

I've reverted any whitespace changes, rebased and squashed. Hopefully this is all good to go. Thanks again for your review.

bjhargrave commented on org.eclipse.transformer.cli/src/test/java/transformer/test/TestTransformClass.java:
```diff
@@ -675,6 +677,98 @@ public void testDirectStrings() throws TransformException, IOException {
+
+        final ClassActionImpl classAction = createOuterClassAction();
+
+        final String resourceName = TEST_DATA_RESOURCE_NAME + '/' + OUTER_CLASS_RESOURCE_NAME;
```
You end up with a double slash here since `OUTER_CLASS_RESOURCE_NAME` starts with a slash.

jgallimore replied:
Thanks! That should be fixed now.

bjhargrave commented:
Thanks!
